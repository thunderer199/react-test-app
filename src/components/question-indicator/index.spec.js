import React from "react";
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer'

import QuestionIndicator from "./index";

describe('QuestionIndicator', () => {

    it('should render right basic example', () => {
        const snapshot = renderer
            .create(<QuestionIndicator questionNumber={1}/>)
            .toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should render right for wrong answered questions', () => {
        const snapshot = renderer
            .create(<QuestionIndicator questionNumber={1} isAnsweredCorrectly={false}/>)
            .toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should render right for right answered questions', () => {
        const snapshot = renderer
            .create(<QuestionIndicator questionNumber={1} isAnsweredCorrectly={true}/>)
            .toJSON();

        expect(snapshot).toMatchSnapshot();
    });

    it('should trigger click when QuestionIndicator clicked', () => {
        const fn = jest.fn();
        const component = shallow(<QuestionIndicator questionNumber={1} click={fn}/>);
        component.simulate('click');

        expect(fn).toHaveBeenCalledTimes(1);
    })
});