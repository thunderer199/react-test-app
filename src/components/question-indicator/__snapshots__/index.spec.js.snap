// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionIndicator should render right basic example 1`] = `
<div
  className="question-indicator question-indicator__unknown"
  onClick={undefined}
>
  <span
    className="question-indicator--text"
  >
    1
  </span>
</div>
`;

exports[`QuestionIndicator should render right for right answered questions 1`] = `
<div
  className="question-indicator question-indicator__right"
  onClick={undefined}
>
  <span
    className="question-indicator--text"
  >
    1
  </span>
</div>
`;

exports[`QuestionIndicator should render right for wrong answered questions 1`] = `
<div
  className="question-indicator question-indicator__wrong"
  onClick={undefined}
>
  <span
    className="question-indicator--text"
  >
    1
  </span>
</div>
`;
